src/pages/distroSettings/DeleteDistro/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+import { useMutation, useQuery } from "@apollo/client";
+import Button from "@leafygreen-ui/button";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Description } from "@leafygreen-ui/typography";
+import { useParams } from "react-router-dom";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import ElementWrapper from "components/SpruceForm/ElementWrapper";
+import { useToastContext } from "context/toast";
+import {
+  DeleteDistroMutation,
+  DeleteDistroMutationVariables,
+  UserDistroSettingsPermissionsQuery,
+  UserDistroSettingsPermissionsQueryVariables,
+} from "gql/generated/types";
+import { DELETE_DISTRO } from "gql/mutations";
+import { USER_DISTRO_SETTINGS_PERMISSIONS } from "gql/queries";
+
+interface ModalProps {
+  closeModal: () => void;
+  distroId: string;
+  open: boolean;
+}
+
+const Modal: React.FC<ModalProps> = ({ closeModal, distroId, open }) => {
+  const dispatchToast = useToastContext();
+
+  const [deleteDistro] = useMutation<
+    DeleteDistroMutation,
+    DeleteDistroMutationVariables
+  >(DELETE_DISTRO, {
+    onCompleted() {
+      dispatchToast.success(
+        `The distro “${distroId}” was deleted. Future visits to this page will result in an error.`
```

Comment:
Would be nice to redirect users to another distro, but I guess it could be hard to ensure it's one they have permissions for. I'm fine with this!